flight/extract_gps.py

```diff
@@ -0,0 +1,94 @@
+"""Contains the extract_gps() function for extracting data out of
+the provided waypoint data JSON file for the SUAS competition."""
+from collections import namedtuple
+from typing import Any, List, NamedTuple
+import json
+from typing_extensions import TypedDict
+
+GPSData = TypedDict(
+    "GPSData",
+    {
+        "waypoints": List[NamedTuple],
+        "boundary_points": List[NamedTuple],
+        "altitude_limits": List[int],
+    },
+)
+
+
+def extract_gps(path: str) -> GPSData:
+    """Returns the waypoints, boundary points, and altitude limits from a waypoint data file.
+
+    Args:
+        path (str): File path to the waypoint data JSON file.
+
+    Returns:
+        GPSData: TypedDict[
+            list(NamedTuple(int, int, int)), list(NamedTuple(int, int)), list(int, int)
+        ]
+            The data in the waypoint data file
+            waypoints : list(NamedTuple(int, int, int))
+                Waypoint : NamedTuple(int, int, int)
+                    latitude : int
+                        The latitude of the waypoint.
+                    longitude : int
+                        The longitude of the waypoint.
+                    altitude : int
+                        The altitude of the waypoint.
+            boundary_points : list(NamedTuple(int, int, int))
+                BoundaryPoint : NamedTuple(int, int, int)
+                    latitude : int
+                        The latitude of the boundary point.
+                    longitude : int
+                        The longitude of the boundary point.
+            altitude_limits : list(int, int)
+                altitude_min : int
+                    The minimum altitude that the drone must fly at all times.
+                altitude_max : int
+                    The maximum altitude that the drone must fly at all times.
+    """
+    # Initialize namedtuples to store latitude/longitude/altitude data for provided points
+    Waypoint = namedtuple("Waypoint", ["latitude", "longitude", "altitude"])
+    BoundaryPoint = namedtuple("BoundaryPoint", ["latitude", "longitude"])
+
+    # Load the JSON file as a Python dict to be able to easily access the data
+    with open(path, encoding="UTF-8") as data_file:
+        json_data: dict[Any, Any] = json.load(data_file)
+
+    # Initialize lists to store waypoints & boundary points
+    waypoints: list[NamedTuple] = []
+    boundary_points: list[NamedTuple] = []
```

Comment:
Once you update Waypoint/BoundaryPoint to classes, you can type as `list[Waypoint]` and `list[BoundaryPoint]` respectively